### PR TITLE
charge better

### PR DESCRIPTION
**Pull request by jr1221**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: Core/Src/stateMachine.c [hunk 1]
```diff
@@ -344,27 +354,70 @@ bool sm_charging_check(acc_data_t *bmsdata)
...
-		//printf("Charger settle countup active\r\n");
-		return false;
+	// if OCV=V, and we are near 4.19V, the formula should not take effect
+	// this is easy to do with a simple one-time trigger, such that if we enter stage 2 we can never go back
+	static bool reached_stage_2 = false;
+
+	// STAGE 1 -- COUNTDOWN=charge, COUNTUP=settle
+	if (bmsdata->max_voltage.val < MAX_CHARGE_VOLT && !reached_stage_2) {
+		// BEGIN COUNTDOWN (first run only)
+		if (!is_timer_expired(&charger_settle_countdown_stage1)) {
```
Comment (caiodasilva2005) [inline Core/Src/stateMachine.c:364]: You should also add a `!is_timer_active()` in this block here because this will infinitely keep restarting the timer since completed will be false and then timer start time will reset

File: Core/Src/stateMachine.c [hunk 2]
```diff
@@ -344,27 +354,70 @@ bool sm_charging_check(acc_data_t *bmsdata)
...
-	// if we are counting down (the normal charging time)
-	if (is_timer_active(&charger_settle_countdown)) {
-		// if we need to stop charging, start the pause timer and stop charging immediately
+	// STAGE 2 & 3 -- COUNTDOWN=charge, COUNTUP=settle
+	if (bmsdata->max_ocv.val < MAX_CHARGE_VOLT) {
+		// mark that stage 2 has been, reached, and we shouldnt go back to stage 1
+		reached_stage_2 = true;
+
+		// BEGIN COUNTDOWN (first run only)
+		if (!is_timer_expired(&charger_settle_countdown)) {
```
Comment (caiodasilva2005) [inline Core/Src/stateMachine.c:394]: same for this one

**Review by caiodasilva2005 — CHANGES_REQUESTED** [review]
